src/libponyrt/lang/stdfd.c

```diff
@@ -438,15 +438,20 @@ PONY_API bool pony_os_stdin_setup()
   // Always use events
   return true;
 #else
-  if(fd_type(STDIN_FILENO) == FD_TYPE_TTY)
+  switch(fd_type(STDIN_FILENO))
   {
-    if(is_stdout_tty)
-      stdin_tty();
-
-    return true;
+    case FD_TYPE_TTY:
+    {
+      if(is_stdout_tty)
+        stdin_tty();
+      return true;
+      break;
```

Comment:
My knowledge of C is quite limited, but I'd expect this line not to be reached. Is it? The return above it isn't conditional afaict...?

Reply:
Yeah, the switch here is weird. I think this is better represented as a if/else if/else series.

Reply:
As you prefer.